fasthttputil/stringslices.go

```diff
@@ -0,0 +1,107 @@
+package fasthttputil
+
+import (
+	"sync"
+
+	"github.com/valyala/bytebufferpool"
+)
+
+var (
+	stringSlicesPool sync.Pool
+)
+
+// AcquireStringSlices returns an empty StringSlice instance from pool.
+//
+// The returned StringSlice instance may be passed to ReleaseStringSlices
+//  when it is no longer needed.
+func AcquireStringSlices() *StringSlices {
+	v := stringSlicesPool.Get()
+	if v == nil {
+		return &StringSlices{}
+	}
+	return v.(*StringSlices)
+}
+
+// ReleaseStringSlices returns slices acquired via AcquireStringSlices to pool.
+func ReleaseStringSlices(slices *StringSlices) {
+	slices.Reset()
+	stringSlicesPool.Put(slices)
+}
+
+// StringSlices is used to convert byte slices to string slices,
+// it only needs to allocate memory once.
+type StringSlices struct {
+	buffer *bytebufferpool.ByteBuffer
+
+	lengths []int
+	pos     int
+	offset  int
+
+	lastError error
+
+	str string
+}
+
+// WriteBytes writes the byte slice of a string.
+func (slices *StringSlices) WriteBytes(bytes []byte) error {
+	if slices.lastError != nil {
+		return nil
+	}
+
+	if slices.buffer == nil {
+		slices.buffer = bytebufferpool.Get()
+	}
+	len, lastError := slices.buffer.Write(bytes)
+	slices.lengths = append(slices.lengths, len)
+	return lastError
+}
+
+// NextStringSlice returns next string slice.
+// it returns empty string if StringSlices have no more string slices.
+func (slices *StringSlices) NextStringSlice() (string, bool) {
```

Comment:
Why the second return value? You seem to be ignoring it everywhere and you already have `Remain()`.